task-3/src/with-time/withTime.ts

```diff
@@ -0,0 +1,17 @@
+import EventEmitter from "../event-emitter/eventEmitter";
+
+class WithTime extends EventEmitter {
+  execute(asyncFunc: (...args: any[]) => Promise<any>, ...args: any[]) {
+    const startTime = Date.now();
+    this.emit("begin");
+
+    asyncFunc(...args).then((data) => {
+      const endTime = Date.now();
+      this.emit("data", data);
+      this.emit("end", endTime - startTime);
+    });
```

Comment:
It's not mentioned in the task description how the system should behave in case of asyncFunc failure, but it would be great to add a catch block here. Starting from node v15 unhandled rejections are thrown to stop the process by default.

Reply:
Fixed, nice catch!
I've read about it, nice to know, thank you :)